src/content/en/fundamentals/getting-started/primers/media-source-extensions.md

```diff
@@ -266,10 +267,14 @@ function sourceOpen(e) {
 
 #### Call endOfStream() 
 
-At the end of processing the media file, `MediaSource.readyState` should be set
-to `ended` and a `sourceended` event should fire. Neither of these happen
-automatically, but we can trigger them by calling `MediaSource.endOfStream()`.
-Do this inside the `SourceBuffer.onupdateend`.
+After all `ArrayBuffers` are appended, and no further media data is expect, call
+`MediaSource.endOfStream()`.  This will change `MediaSource.readyState` to
+`ended` and fire the `sourceended` event. 
+
+You also need to disconnect the blob URL from the `MediaSource` object. This
+allows the browser to reclaim the resources alocated to the `MediaSource`
+instance, assuming there are no references to it elsewhere. Do this by calling
+`revokeObjectURL()` on the source itself.
 
 <pre class="prettyprint">
 function sourceOpen(e) {
```

Comment:
Switch to `var mediaSource = e.target` below and all other places.
Look for `this;` and replace three remaining occurrences.